skip_message_sending_on_holidays.py

```diff
@@ -0,0 +1,231 @@
+            value=quote_author
+        )
+    except (requests.RequestException, ValueError) as e:
+        logging.error(f"Error while loading quote: {e}")
+        raise
+
+
+def load_image(ti):
+    try:
+        image_generator = ImageGeneratorFromPiscumAPI()
+        image_url = image_generator.get_content()
+        if not image_url:
+            logging.error("Failed to generate image.")
+            raise ValueError("Failed to generate image.")
+        ti.xcom_push(
+            key='image_url',
+            value=image_url
+        )
+    except (requests.RequestException, ValueError) as e:
+        logging.error(f"Error while loading image: {e}")
+        raise
+
+
+def send_message(ti):
+    try:
+        quote_text = ti.xcom_pull(
+            key='quote_text',
+            task_ids='load_quote'
+        )
+        quote_author = ti.xcom_pull(
+            key='quote_author',
+            task_ids='load_quote'
+        )
+        image_url = ti.xcom_pull(
+            key='image_url',
+            task_ids='load_image'
+        )
+        if not quote_text or not quote_author or not image_url:
+            logging.error("Failed to send message due to missing content")
+            raise ValueError("Missing content for sending message.")
+
+        message_sender = TeamsMessageSender(TEAMS_WEBHOOK_URL)
+        message_sender.send_message(
+            "Sent by Natalia Ananeva",
+            quote_text,
+            quote_author,
+            image_url
+        )
+
+    except (requests.RequestException, ValueError) as e:
+        logging.error(f"Error while sending message: {e}")
+        raise
+
+
+DATES_TO_SKIP = [
+    datetime.datetime(2024, 6, 11),
+    datetime.datetime(2024, 6, 13),
+    datetime.datetime(2024, 6, 15),
+    datetime.datetime(2024, 6, 25),
+]
+
+
+def check_date(**kwargs):
+    """
+    Function checks whether current execution date is in the lists of
+    dates which should be skipped.
+    """
+    execution_date = kwargs['execution_date']
+    if execution_date in DATES_TO_SKIP:
+        return 'finish'  # Task to go to if the date is excluded
+    return ['load_quote_op', 'load_image_op']  # Task to go to if the date is not excluded
+
+
+with DAG(
+    dag_id='send_massage_msteams_skip_holidays',
+    start_date=datetime.datetime(
+        year=2024,
+        month=6,
+        day=1
+    ),
+    schedule_interval=timedelta(days=1),
+    tags=['send_message_teams'],
+    description="A DAG to send message to MS Teams channel using webhook. \
+                 Message consists of quote and picture. \
+                 In this version some holidays should be skipped.",
+    catchup=False
+) as dag:
+
+    start_op = EmptyOperator(
+        task_id='start'
+    )
+
+    check_date_op = BranchPythonOperator(
+        task_id='check_date',
+        python_callable=skip_if_excluded,
+        provide_context=True,  # is used to pass Airflow's context variables to the Python callable
```

Comment:
This parameter is redundant in the recent versions of Airflow

Reply:
deleted

thanks!